_posts/en/newsletters/2020-12-09-newsletter.md

```diff
@@ -0,0 +1,174 @@
+  unchanged [BIP173][] address scheme.  Based on his research and the
+  other information available, and if there are no convincing
+  objections, he plans to write a BIP for the tweaked bech32 checksum
+  formula to be used for segwit v1 addresses (taproot) and subsequent
+  segwit versioned addresses.  If the proposed alternative bech32
+  addresses are adopted, the following implications exist for wallets
+  and services:
+
+    - *Support for current addresses remains unchanged:* wallets that
+      can send to current native segwit addresses (version 0), receive
+      to current segwit addresses, or do both will continue to work
+      without modification.  Current native segwit addresses all start
+      with `bc1q`.  P2SH-wrapped segwit addresses are also unaffected.
+
+    - *Wallets that support segwit v1+ addresses now won't be forward compatible:*
+      wallets that implemented BIP173 forward compatibility will find it
+      doesn't work with taproot and subsequent addresses generated using
+      Wuille's new proposal.  Those addresses will fail with a checksum
+      mismatch.  According to the survey results posted to the mailing
+      list, the only known wallets affected are Bitcoin Core and BRD
+      wallet.
+
+    - *All other wallets will need to update eventually anyway:* the
+      other wallets and services surveyed don't currently support
+      planned taproot addresses, so they'll need to upgrade anyway when
+      users start requesting money be sent to those v1 addresses.  For
+      any wallets that already support sending to current segwit
+      addresses, the update should be easy as the planned algorithm
+      change is minimal.
+
+- **Thwarted upgrade to uppercase bech32 QR codes:** BTCPay Server
+  recently implemented an optimization described in [BIP173][] (covered
+  in [Newsletter #46][bech32 upper]) of creating QR codes for bech32
+  addresses that [start with][bip21 lowercase] all capital letters in order to create
+  smaller and less complex QR codes.  Unfortunately, a merchant using
+  BTCPay soon [reported][btcpay #2099] that one of their customers was
+  unable to pay the new QR codes.  With impressive speed, contributors
+  to BTCPay soon found a popular wallet that did not properly implement
+  [RFC3986][] case insensitive schema parsing, and so BTCPay developers
+  reverted the QR code change.  Other wallets that also generate
+  uppercase QR codes, such as Bitcoin Core since version 0.19, still
+  continue to do so.  It would be beneficial for all users if any other
+  wallets that don't accept uppercase QR codes could be identified and a
+  polite bug report filed with their developers so that everyone can use
+  maximally efficient QR encoded invoice strings in the future.  To that
+  end, you may wish to test your preferred wallets or services with the
+  following examples (scan the codes, but please don't send money).
+
+    FIXME:QR codes
```

Comment:
What's the plan here? A black hole address? Anyone-can-spend? An address controlled by you so you can return funds accidentally sent there?

Reply:
I was going to use an address controlled by me, but I can do `p2wsh(OP_TRUE)` if you want.

Reply:
> I was going to use an address controlled by me

This seems reasonable. At least then there's a chance of returning the funds to anyone who accidentally sends to the adddress.

Reply:
Actually, it looks like I'm going to need to rewrite this section a fair bit; the BTCPay team is waaaaay ahead of me on this and have already performed a significant amount of testing.